sunpy/net/dataretriever/sources/rhessi.py

```diff
@@ -193,9 +192,23 @@ def _can_handle_query(cls, *query):
         boolean
             answer as to whether client can service the query
         """
-        chkattr = ['Time', 'Instrument']
-        chklist = [x.__class__.__name__ in chkattr for x in query]
+        from sunpy.net import attrs as a
+
+        required = {a.Time, a.Instrument}
+        optional = {a.Physobs}
+
+        all_attrs = {type(x) for x in query}
+
+        query_attrs = set(type(x) for x in query)
+
+        if not required.issubset(query_attrs) or not all_attrs.issubset(query_attrs):
+            return False
+
+        matches = True
```

Comment:
Any reason to match true first?

Reply:
It's just the way I wrote the logic, could just as easily be the other way around :man_shrugging:

Reply:
Ignore me then.